core/script.runtime/src/kotlin/script/dependencies/resolvers.kt

```diff
@@ -47,18 +66,23 @@ interface DependenciesResolver : @Suppress("DEPRECATION") ScriptDependenciesReso
     }
 }
 
+// is File a could type to use here?
+// No way to get script name if file is not present, should add another property (val fileName: String)
 interface ScriptContents {
     val file: File?
     val annotations: Iterable<Annotation>
     val text: CharSequence?
```

Comment:
I believe it's easier for the client code to work with this text as a string, there's no real reason for the implementor to return any other char sequence except string, and `CharSequence`s are rarely found in good APIs anyway.